### PR TITLE
Improve running mode configs

### LINKED ISSUE #1269: Improve the usage of the running mode config file
### Context
After discussion, we'd like to:

- merge different running modes (`mock`, `dev`, `staging`, `prod`) into one single config file: `running-mode-config.json`
- introduce an env for each config field (e.g. TWITTER_OFFICIAL_URL), use it when it's present
- if no env is present, fallback to the values in `running-mode-config.json`
- (optional) add an extra arg to use a specified running-mode config file 

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by ziming-zung**
This PR resolves #1269

### CODE REVIEW

File: tee-worker/service/src/main.rs [hunk 1]
```diff
@@ -872,3 +828,47 @@ fn check_we_are_primary_validateer(
 		node_api.enclave_count(Some(*register_enclave_xt_header.parent_hash()))?;
 	Ok(enclave_count_of_previous_block == 0)
 }
+
+fn data_provider(config: &Config) -> DataProvidersStatic {
+	let build_in_mode = vec!["dev", "mock", "prod", "staging"];
```
Comment (Kailai-Wang) [inline tee-worker/service/src/main.rs]: maybe you mean `built_in_mode`?

File: tee-worker/service/src/main.rs [hunk 2]
```diff
@@ -872,3 +828,47 @@ fn check_we_are_primary_validateer(
...
+	let build_in_config: Value =
+		serde_json::from_slice(include_bytes!("running-mode-config.json")).unwrap();
+
+	let mut data_provider_config = if build_in_mode.contains(&config.running_mode.as_str()) {
```
Comment (Kailai-Wang) [inline tee-worker/service/src/main.rs]: So if I understand it correctly, `running-mode` should be either one of the built-in modes, or a file path?

File: tee-worker/service/src/main.rs [hunk 3]
```diff
@@ -872,3 +828,47 @@ fn check_we_are_primary_validateer(
...
+	} else {
+		let file_path = config.running_mode.as_str();
+		let mut file = File::open(file_path)
+			.map_err(|e| format!("{:?}, file:{}", e, file_path))
+			.unwrap();
+		let mut data = String::new();
+		file.read_to_string(&mut data).unwrap();
+		serde_json::from_str::<DataProvidersStatic>(data.as_str()).unwrap()
+	};
+	if let Ok(v) = env::var("TWITTER_OFFICIAL_URL") {
```
Comment (Kailai-Wang) [inline tee-worker/service/src/main.rs:849]: A small question is if all `env`s are set but the running-mode is incorrectly configured, should we still allow this? Current logic is we would panic out before checking the env.

Not a big deal though

**Review by Kailai-Wang — APPROVED** [review]
Looks good, thanks, just a few small questions from my side

File: tee-worker/service/src/main.rs (hunk 1 above)
Comment (ziming-zung) [reply tee-worker/service/src/main.rs]: Yes, it shoud be `built_in_mode` :rofl:

File: tee-worker/service/src/main.rs (hunk 2 above)
Comment (ziming-zung) [reply tee-worker/service/src/main.rs]: Yes

File: tee-worker/service/src/main.rs (hunk 3 above)
Comment (ziming-zung) [reply tee-worker/service/src/main.rs:849]: I think `built_in_config` or config from file_path should always be correct. :sweat_smile:

**Pull request merged**